.changeset/dull-foxes-sit.md

```
@@ -0,0 +1,5 @@
+---
+'@mastra/deployer': patch
+---
+
+Fixed deployer analysis to use one deterministic external dependency set across analysis, bundling, and validation. Deprecated externals such as `nodemailer`, `jsdom`, `sqlite3`, and `fastembed` are now declared and installed as runtime dependencies instead of bundled. This can cause a one-time experiment-worker digest and build-key change for affected projects.
```

markdownlint-cli2 (0.23.2)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)